### PR TITLE
Update MariaDB client driver to version 2

### PR DESCRIPTION
**Pull request by uncleSlayer**
## Description
Updated mariadb driver to version 2

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
image
Runs 10 conversations parallely without taking the query service down

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/sources/client/mariadb/mariadb.py [hunk 1]
```diff
@@ -140,208 +118,142 @@ def connect(self) -> "MariaDBClient":
...
-                    pool_name=self._pool_name,
-                    pool_size=self.pool_size,
-                    pool_reset_connection=True,
+                self._pool = await mariadb.create_async_pool(
+                    min_size=1,
+                    max_size=self.pool_size,
+                    acquire_timeout=self.pool_acquire_timeout,
+                    reset_connection=True,
                     **connect_kwargs,
                 )
```
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py:127]: Wrong review

File: backend/python/app/sources/client/mariadb/mariadb.py [hunk 2]
```diff
@@ -140,208 +118,142 @@ def connect(self) -> "MariaDBClient":
...
-
-            conn.commit()
-            return results
-
-        except mariadb.Error as e:
-            try:
-                conn.rollback()
-            except mariadb.Error:
-                pass
+            async with self._pool.connection(timeout=self.pool_acquire_timeout) as conn:
```
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py]: will investigate

File: backend/python/app/sources/client/mariadb/mariadb.py [hunk 3]
```diff
@@ -140,208 +118,142 @@ def connect(self) -> "MariaDBClient":
...
+        if self._pool is None:
+            raise RuntimeError("MariaDB pool not initialized")
 
-        except mariadb.Error as e:
-            try:
-                conn.rollback()
-            except mariadb.Error:
-                pass
+        try:
+            async with self._pool.connection(timeout=self.pool_acquire_timeout) as conn:
```
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py]: removed timeout

File: backend/python/app/sources/client/mariadb/mariadb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py:127]: wrong review

File: backend/python/app/sources/client/mariadb/mariadb.py [hunk 4]
```diff
@@ -140,208 +118,142 @@ def connect(self) -> "MariaDBClient":
...
-
-        except mariadb.Error as e:
-            try:
-                conn.rollback()
-            except mariadb.Error:
-                pass
+            async with self._pool.connection(timeout=self.pool_acquire_timeout) as conn:
+                cursor = None
+                try:
+                    cursor = conn.cursor(dictionary=True)
```
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py:179]: Wrong review, conn.cursor is synchronous.
cursor.execute is async instead so so we can do await there.

File: backend/python/app/sources/client/mariadb/mariadb.py [hunk 5]
```diff
@@ -140,208 +118,142 @@ def connect(self) -> "MariaDBClient":
...
-        except mariadb.Error as e:
-            try:
-                conn.rollback()
-            except mariadb.Error:
-                pass
+        try:
+            async with self._pool.connection(timeout=self.pool_acquire_timeout) as conn:
+                cursor = None
+                try:
+                    cursor = conn.cursor()
```
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py:222]: Wrong review, conn.cursor is synchronous.
cursor.execute is async instead so so we can do await there.

File: backend/python/app/sources/client/mariadb/mariadb.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (siddhant-ota) [reply backend/python/app/sources/client/mariadb/mariadb.py]: fixed

**Review by abhishek-pipeshub — APPROVED** [review]
